main.cpp

```diff
@@ -198,44 +197,49 @@ static int parse_argv(int argc, char *argv[], const char **uart, char **udp_addr
     }
 
     /* positional arguments */
-    if (optind + 1 != argc) {
+    if (optind + 1 > argc) {
         log_error("Error parsing required argument %d %d", optind, argc);
         help(stderr);
         return -EINVAL;
     }
 
-    if (stat(argv[optind], &st) == -1 || !S_ISCHR(st.st_mode)) {
-        char *ip = strdup(argv[optind]);
+    while (optind < argc) {
+        if (stat(argv[optind], &st) == -1 || !S_ISCHR(st.st_mode)) {
+            unsigned long port;
+            char *ip = strdup(argv[optind]);
 
-        char *portstr = strchrnul(ip, ':');
-        if (*portstr == '\0') {
-            log_error("Invalid argument for UDP address = %s. Please inform <address>:<port>", argv[optind]);
-            help(stderr);
-            free(ip);
-            return -EINVAL;
-        } else {
-            *portstr = '\0';
-            if (safe_atoul(portstr + 1, udp_port) < 0) {
-                log_error("Invalid argument for UDP port = %s", argv[optind]);
+            char *portstr = strchrnul(ip, ':');
+            if (*portstr == '\0') {
+                log_error("Invalid argument for UDP address = %s. Please inform <address>:<port>",
+                          argv[optind]);
                 help(stderr);
                 free(ip);
                 return -EINVAL;
+            } else {
+                *portstr = '\0';
+                if (safe_atoul(portstr + 1, &port) < 0) {
+                    log_error("Invalid argument for UDP port = %s", argv[optind]);
+                    help(stderr);
+                    free(ip);
+                    return -EINVAL;
+                }
             }
-        }
-        *udp_addr = ip;
-    } else {
-        *uart = argv[optind];
-    }
 
-    /* Will route from UART xor UDP */
-    if (*uart && *udp_port) {
-        log_error("Cannot route from UART and UDP at same time!");
-        help(stderr);
-        return -EINVAL;
-    } else if (!*uart && !*udp_port) {
-        log_error("Please, inform a UART device or UDP port to route!");
-        help(stderr);
-        return -EINVAL;
+            struct endpoint_address *e = (struct endpoint_address *)malloc(sizeof(*e));
+            e->next = opt.master_addrs;
+            e->ip = ip;
+            e->port = port;
```

Comment:
endpoint_address could be a class that receives ip and port and later sets its position in the list...
or at least have a function `create_endpoint_adress()`... makes it easier to handle the ownership of objects and avoids leaks.